addons/web/static/src/js/core/class_patcher.js

```diff
@@ -0,0 +1,48 @@
+odoo.define("web.class_patcher", function () {
+    "use strict";
+
+    function makePatchable(OriginalClass) {
+        let unpatchList = [];
+        class PatchableClass extends OriginalClass {}
+
+        PatchableClass.patch = function(name, patch) {
```

Comment:
I think the very edge case where the patch arrives after the rendering of the orignal owl component is not supported.
What I started doing is https://github.com/odoo-dev/enterprise/commit/60a4ccf5179c98b0e8d571ad8632bd5279b6edf0

In the case we add a component, both in the template and in the component key of the class, a instance member should hold constructor.component, a if is added in the template, a bus event is trigger to tell the already instanciated/rendered component to re-render.

All in all all that stuff is unnatural. In my case though, the webClient, it is the very first component to be instanciated and overridden, so maybe my case is too edgy and won't occur otherwise